lib/sles4sap/microsoft_sdaf.pm

```diff
@@ -0,0 +1,177 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+#
+# Library used for Microsoft SDAF deployment
+
+package sles4sap::microsoft_sdaf;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use mmapi qw(get_current_job_id);
+use utils qw(write_sut_file file_content_replace);
+use qesapdeployment qw(qesap_az_get_resource_group);
+use File::Basename;
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation:
+https://learn.microsoft.com/en-us/azure/sap/automation/get-started
+
+=cut
+
+our @EXPORT = qw(
+  az_login
+  sdaf_prepare_ssh_keys
+  sdaf_get_deployer_ip
+);
+
+=head2 az_login
+
+ az_login();
+
+Azure login using SPN credentials defined by secret OpenQA parameters:
+
+B<_SECRET_AZURE_SPN_APPLICATION_ID>
+
+B<_SECRET_AZURE_SPN_APP_PASSWORD>
+
+B<_SECRET_AZURE_ARM_TENANT_ID>
+
+Returns 'subscription ID' on success.
+
+=cut
+
+sub az_login {
+    # Note: For login I cannot use standard PC library, because it gives credentials which are missing permissions
+    my $app_id = get_required_var('_SECRET_AZURE_SDAF_APP_ID');
```

Comment:
Atm, PC library is doing more than SDAF needs and I don't want to go into that rabbit hole at this stage of development since the deadline is slowly closing. But I was talking already with Michele about this topic, so I am keeping it in mind.